File: src/1805.number-of-different-integers-in-a-string.1933/solution.js
/**
 * @param {string} word
 * @return {number}
 */
var numDifferentIntegers = function (word) {
    const set = new Set();
    let index = 0;
    while (index < word.length) {
        if (!isDigit(word.charCodeAt(index))) {
            index++;
            continue;
        }
        const start = index++;
        while (index < word.length && isDigit(word.charCodeAt(index))) {
            index++;
        }
        set.add(+word.slice(start, index));
    }
    return set.size;
};

function isDigit (code) {
    return code > 47 && code < 58;
}

File: src/1805.number-of-different-integers-in-a-string.1933/solution.java
class Solution {
    public int numDifferentIntegers(String word) {
        word = word.replaceAll("\\D+", " ").trim();
        if(word.isEmpty()){
            return 0;
        }
        
        return (int) Arrays.stream(word.split("\\s+")).map(this::trimLeading0).distinct().count();
    }

    private String trimLeading0(String s){
        int index = 0;
        char[] chars = s.toCharArray();
        while (index<chars.length && chars[index] == '0'){
            index++;
        }
        if(index == chars.length){
            return "0";
        }
        return s.substring(index);
    }
}